File: library/src/main/java/me/proxer/library/api/Endpoint.java
package me.proxer.library.api;

import javax.annotation.Nonnull;

/**
 * Interface for all endpoints.
 *
 * @param <T> The type this endpoint returns.
 * @author Ruben Gees
 */
public interface Endpoint<T> {

    /**
     * Builds the call for this endpoint.
     */
    @Nonnull
    ProxerCall<T> build();
}

File: library/src/main/kotlin/me/proxer/library/api/Endpoint.kt
package me.proxer.library.api

/**
 * Interface for all endpoints.
 *
 * @param <T> The type this endpoint returns.
 *
 * @author Ruben Gees
 */
interface Endpoint<T> {

    /**
     * Builds the call for this endpoint.
     */
    fun build(): ProxerCall<T>
}
